### PR TITLE
Ye another windows issues

### PR DESCRIPTION
**Pull request by aleksandrm8**
Hello!
Fix here some windows issues (Visual Studio 2015).
Regards.

### CODE REVIEW

File: include/nmealib/sentence.h [hunk 1]
```diff
@@ -124,6 +125,32 @@ static const NmeaSentencePrefixToType nmealibSentencePrefixToType[] = {
         .prefix = NULL, //
         .sentence = NMEALIB_SENTENCE_GPNON //
     }//
+#else
+    {
```
Comment (fhuberts) [inline include/nmealib/sentence.h]: really? vs can't handle this?
did they just slap a 2015 label on a 1990 compiler or what?

If so, then I'd like this to not be windows specific but generic.

Comment (fhuberts) [reply include/nmealib/sentence.h]: Can you change compiler settings to C99, then it should work

**Comment by fhuberts** [comment]
please try setting the compiler to C99, then the changes to the struct initialisers are not needed

File: include/nmealib/sentence.h (hunk 1 above)
Comment (aleksandrm8) [reply include/nmealib/sentence.h]: You are right. But VS can not compile *.cpp file with C compiler. So, it is necessary to make a C wrapper to use nmealib in c++ project. But maybe it is better to apply patch and make nmealib compatible for both c and c++ ? Another solution is to make NmeaSentencePrefixToType "extern" (move definition to sentence.c). I tested - it works like a charm. What do you think about this offer?

**Comment by aleksandrm8** [comment]
You are right. But VS can not compile *.cpp file with C compiler. So, it is necessary to make a C wrapper to use nmealib in c++ project. But maybe it is better to apply patch and make nmealib compatible for both c and c++ ? Another solution is to make NmeaSentencePrefixToType "extern" (move definition to sentence.c). I tested - it works like a charm. What do you think about this offer?

**Comment by aleksandrm8** [comment]
@fhuberts , what do you think about my post above?

**Comment by fhuberts** [comment]
>You are right. But VS can not compile *.cpp file with C compiler

What are you talking about? There are no cpp files in nmealib.
Also, all headers have
```
#ifdef  __cplusplus
extern "C" {
#endif /* __cplusplus */

...

#ifdef  __cplusplus
}
#endif /* __cplusplus */
```

**Comment by aleksandrm8** [comment]
@fhuberts , you are right. There are no cpp files in nmealib. I am talking about using nmealib in cpp project. So when nmealib used in c++ project, c++ compiler is used. And c++ compiler (at least Visual Studio 2015 **c++ compiler**) does not support designated structure initialization (because it is C feature). The only one opportunity to use nmealib inside VS c++ project is to use additional files with "*.c" extension inside c++ project (e.g. C wrapper). I am offer avoid it by applying this patch or by moving NmeaSentencePrefixToType struct initialization to "src/sentence.c" file (and leave extern declaration in sentence.h).

@fhuberts , extern "C" just change linkage type, not more.
Regards.

**Comment by fhuberts** [comment]
ok, so I've loaded in VS2015 up what you did and it's far from complete.
There are many more things you need to fix before windows truly is supported, for example: the nmeaRandom function.
I'm going to revert your previous PR and will only merge a windows support PR again when the support is done and full.

**Pull request closed without merging**